app/models/katello/content_view_package_filter.rb

```diff
@@ -7,10 +7,14 @@ class ContentViewPackageFilter < ContentViewFilter
     validates_lengths_from_database
 
     # Returns a set of Pulp/MongoDB conditions to filter out packages in the
-    # repo repository that match parameters
+    # repo repository that match parameters if the repository content type
+    # is managed by pulp2.
+    #
+    # Returns a set of pulpcore hrefs (pulp_ids) if the repository content
+    # type is managed by pulpcore.
     #
     # @param repo [Repository] a repository containing packages to filter
-    # @return [Array] an array of hashes with MongoDB conditions
+    # @return [Array] an array of hashes with MongoDB conditions or pulpcore hrefs
```

Comment:
I'm a little confused here, how would `generate_clauses` return pulpcore hrefs?

Reply:
That comment was left over from a very early revision, before I realized that overloading the existing methods with pulpcore specific support wasn't a good idea. It's misleading so thank you for noticing it!